docs/csharp-api/index.md

```diff
@@ -0,0 +1,24 @@
+# Industrial IoT Platform C# Reference
```

Comment:
Q: Why not put this under api/ folder?

Reply:
Moved under api folder, which is the better place